### ISSUE TITLE
Memory leak test

### ISSUE THREAD

**Issue by samreid** [phetsims/circuit-construction-kit-common#71]
From https://github.com/phetsims/tasks/issues/546 we should check memory leaks in this sim.

**Comment by samreid** [comment]
* launch sim with fuzzMouse=100
* wait 30 seconds
* take 1st heap snapshot
* wait 15 seconds
* take 2nd heap snapshot

There is a serious leak:
![image](https://cloud.githubusercontent.com/assets/679486/14696698/87a42404-073a-11e6-8217-f25d931bedac.png)

Approximately leaking 6MB/sec 😟

**Comment by samreid** [comment]
The fixes above reduce the severity of the leak significantly, but we still have a ways to go.

**Comment by samreid** [comment]
I'm having trouble making progress here because there is so much scenery noise, which I am not sure whether my application is responsible for or not.  In particular, many allocations still have a retainer through `previousFirstDrawable` and `previousLastDrawable`

**Comment by samreid** [comment]
I've addressed many leak issues above but it is difficult to proceed while https://github.com/phetsims/scenery/issues/542 is a problem.  I'll put this issue on hold until we can discuss the scenery issue with @jonathanolson

**Comment by samreid** [comment]
The scenery issue may be a red herring.  Subsequent allocations don't show as leaks.  Reopening for continued investigation.

**Comment by samreid** [comment]
I see 28MB snapshot after starting screen 1 (not doing anything).  This seems normal so far. After creating a small circuit, it is at 55MB.  Then after reset all, still 55MB.

@phet-steele says he is up to 500MB and a very hot CPU.

**Comment by samreid** [comment]
Fixed a memory leak for ElectronNode above.

**Comment by samreid** [comment]
After 60 seconds at fuzzMouse=100, I'm seeing the memory usage go from 31MB to 56MB, so there is still work to be done here.

**Comment by samreid** [comment]
After the above commit, we are leaking 3MB/minute. Still some ways from the "acceptable 0.1MB/minute" range, but getting closer.

**Comment by samreid** [comment]
I ran a larger (5 minute) test, sampling once per minute, running Canary and ?screens=1.

I saw:
0 min: 25MB
1 min: 30.5MB
2 min: 26.3MB
3 min: 30.7MB
4 min: 30.1MB
5 min: 29.8MB

![image](https://cloud.githubusercontent.com/assets/679486/16784173/c40a7268-4844-11e6-85c8-ff5b8ef3d6db.png)

This suggests that the problem is resolved.  I should note there is noise in taking memory snapshots from CCK using fuzzMouse because the number of created items varies.

This seems good enough to close, and we'll review again before publishing DC.
